hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -380,6 +387,24 @@ else if ( enhancementContext.doExtendedEnhancement( managedCtClass ) ) {
 		}
 	}
 
+	// See HHH-16572
+	// return true if enhancement is unsupported
+	private boolean unsupportedEnhancement(TypeDescription managedCtClass) {
+		MethodList<MethodDescription.InDefinedShape> list = managedCtClass.getDeclaredMethods();
+		for ( MethodDescription.InDefinedShape shape: list) {
+			AnnotationList annotationList = shape.getDeclaredAnnotations();
+
+			if ( annotationList.isAnnotationPresent(Id.class)) {
+				log.debugf( "Skipping enhancement of [%s]: due to use of [%s] annotation", managedCtClass.getName(), Id.class.getName() );
+				return true;
+			} else if ( annotationList.isAnnotationPresent(Access.class)) {
```

Comment:
The Access.class check is wrong and needs to instead check for @Access(PROPERTY) specifically.